internal/ledgerutil/compare.go

```diff
@@ -39,8 +44,20 @@ func Compare(snapshotDir1 string, snapshotDir2 string, outputFile string) (count
 		return 0, errors.New("both snapshots public state hashes are same. Aborting compare")
 	}
 
+	// Output directory creation
+	outputDirName := filepath.Base(snapshotDir1) + "_and_" + filepath.Base(snapshotDir2) + "_compared"
+	outputDirPath := filepath.Join(outputDirLoc, outputDirName)
+
+	empty, err := fileutil.CreateDirIfMissing(outputDirPath)
+	if err != nil {
+		return 0, err
+	}
+	if !empty {
+		return 0, errors.Errorf("%s already exists in %s. Choose a different location or remove the existing results. Aborting compare", outputDirName, outputDirLoc)
```

Comment:
If user doesn't pass and output directory, this leads to an awkward message like
"4_and_4_compared already exists in . Choose a different location or remove the existing results."

Reply:
This is also what used to happen if the command was successful. Manish felt it wasn't necessary to focus on those edge cases but I can do a quick fix.